### DISCUSSION TITLE
How do your Cloudflare bypass Work?

### DISCUSSION THREAD

**Discussion by sajedulsakib001** [kodjodevf/mangayomi#29] [Ideas]
I made an attempt to figure it out for one of my html parsing project, but since I lack any experience with Flutter, I found it quite challenging. Could you please consider setting up a separate repository specifically focused on bypassing Cloudflare using Flutter? I did try to implement this functionality into a project using your original code, but due to my limited knowledge of Flutter and the isar/isar_flutter_libs libraries, I couldn't figure out how to properly store and retrieve cookies from storage. I also made an attempt to use an alternative JSON-based storage library called 'localstorage' because I found it easier to understand. However, it seems like not functioning as expected. I would greatly appreciate your help with this matter.

**Comment by kodjodevf** [comment]
Unfortunately this no longer works, I'm also trying to find another way perhaps through platform channels using native

**Comment by sajedulsakib001** [comment]
I just use this simple code to save cookie in shared_preferences... and working fine I also saved useragent  to use this cookie with http request so that I don't have to call headlesswebview every time..
````
CookieMain().setCookie(id, encoded);
````
It's calling a function that just save cookie in shared_preferences  with id that's it...

![Screenshot_2023-10-02-20-28-21-890_com android chrome-edit](https://github.com/kodjodevf/mangayomi/assets/48612479/cb8699a4-4c36-4f70-b6c7-bf79f4e7a16e)

**Reply by kodjodevf** [reply]
Nice, can you make a pull request ?

**Reply by sajedulsakib001** [reply]
Yeah, no problem, but it may take some time.

**Reply by sajedulsakib001** [reply]
@kodjodevf Did you fixed the problem

**Reply by sajedulsakib001** [reply]
I'm sorry, bro. I was away from home for almost a month.

**Reply by kodjodevf** [reply]
I understand, I haven't fixed it yet.

**Comment by sajedulsakib001** [comment]
This is what i did to just bypass cloudflare


[cloudflare_bypass.zip](https://github.com/kodjodevf/mangayomi/files/13197314/cloudflare_bypass.zip)

**Reply by kodjodevf** [reply]
Will test it thanks
